src/plugins/crypto/crypto.c

```diff
@@ -205,15 +205,35 @@ int CRYPTO_PLUGIN_FUNCTION (open) (Plugin * handle ELEKTRA_UNUSED, Key * errorKe
  * @retval 1 on success
  * @retval -1 on failure
  */
-int CRYPTO_PLUGIN_FUNCTION (close) (Plugin * handle ELEKTRA_UNUSED, Key * errorKey ELEKTRA_UNUSED)
+int CRYPTO_PLUGIN_FUNCTION (close) (Plugin * handle, Key * errorKey ELEKTRA_UNUSED)
 {
+	/* default behaviour: no teardown except the user/system requests it */
+	KeySet * pluginConfig = elektraPluginGetConfig (handle);
+	if (!pluginConfig)
+	{
+		return -1; // failure because of missing plugin config
+	}
+
+	Key * shutdown = ksLookupByName (pluginConfig, ELEKTRA_CRYPTO_PARAM_SHUTDOWN, 0);
+	if (!shutdown)
+	{
+		return 1; // applying default behaviour -> success
+	}
+	else
+	{
+		if (strcmp (keyString (shutdown), "true") != 0)
```

Comment:
The python plugin uses a ref counter (protected by a mutex). The crypto plugin does actually the same thing.

There is no mentioning of "python" in `doc/decisions`.